### PR TITLE
Change kube2sky to use token-system-dns secret, point at https endpoint ...

### PR DESCRIPTION
**Pull request by cjcullen**
...(instead of kubernetes-ro service).

### CODE REVIEW

**Comment by cjcullen** [comment]
After I get an LGTM and a clean e2e, I'll rebuild the kube2sky:1.2 image and push before merging.

File: cluster/addons/dns/skydns-rc.yaml.in [hunk 1]
```diff
@@ -46,3 +51,11 @@ desiredState:
               - name: dns
                 containerPort: 53
                 protocol: UDP
+        volumes:
+          - name: dns-token
+            source:
+              secret:
+                target:
+                  kind: Secret
```
Comment (thockin) [inline cluster/addons/dns/skydns-rc.yaml.in]: @pmorie have we fixed or disproved the Secret idempotency problem?

File: cluster/addons/dns/skydns-rc.yaml.in [hunk 2]
```diff
@@ -29,10 +29,15 @@ desiredState:
                     "-advertise-client-urls=http://127.0.0.1:4001",
             ]
           - name: kube2sky
-            image: gcr.io/google_containers/kube2sky:1.1
+            image: gcr.io/google-containers/kube2sky:1.2
```
Comment (thockin) [inline cluster/addons/dns/skydns-rc.yaml.in]: did you already push this?

File: cluster/addons/dns/kube2sky/kube2sky.go [hunk 3]
```diff
@@ -128,22 +130,31 @@ func newEtcdClient() (client *etcd.Client) {
...
+		if masterPort == "" {
+			log.Fatalf("KUBERNETES_RO_SERVICE_PORT is not defined")
+		}
+		config = &kclient.Config{
+			Host:    fmt.Sprintf("http://%s:%s", masterHost, masterPort),
+			Version: "v1beta1",
+		}
+	} else {
+		var err error
+		if config, err = kclientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```
Comment (roberthbailey) [inline cluster/addons/dns/kube2sky/kube2sky.go]: Will this use the v1beta3 API?

**Comment by thockin** [comment]
LGTM but question to @pmorie re: secrets

kube2sky needs a near-total overhaul, but that should not stop this PR.

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 2 above)
Comment (roberthbailey) [inline cluster/addons/dns/skydns-rc.yaml.in]: google_containers here so it works on docker versions older than 1.5

File: cluster/addons/dns/kube2sky/kube2sky.go (hunk 3 above)
Comment (cjcullen) [inline cluster/addons/dns/kube2sky/kube2sky.go]: Changing to v1beta3 required other changes. I'd rather see that go in a different PR.

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 2 above)
Comment (cjcullen) [inline cluster/addons/dns/skydns-rc.yaml.in]: Yes, already pushed (hopefully nobody is using latest?).
And fixed the _ vs. -

Comment (pmorie) [inline cluster/addons/dns/skydns-rc.yaml.in]: @thockin it needs to be fix and is pn my list for today

Comment (pmorie) [inline cluster/addons/dns/skydns-rc.yaml.in]: See #7166

**Comment by cjcullen** [comment]
E2Es are green except the load balancer test (I haven't fetched/rebased since last night when LBs were broken). I think this is fine to merge...

**Comment by quinton-hoole** [comment]
This merge appears to have significantly increased flakyness of DNS e2e tests (see #7453 for details).  Broken cluster DNS in turn causes a few other e2e tests which rely on DNS to fail.

I'm going to test that a rollback improves things, and then do it.

**Comment by cjcullen** [comment]
This PR had already been rolled back due to a conflict with another PR. You are probably looking for #7275. I've put together #7461 to roll-back the version of kube2sky to the non-flaky one.

**Comment by jimjh** [comment]
I ran into some problems with the `token-system-dns` (see #7359), which I worked around by creating an empty secret.
